### PR TITLE
Allow to stop cache without management queue

### PR DESCRIPTION
**Pull request by arutk**

### CODE REVIEW

**Comment by arutk** [comment]
This change will be integrated after 19.6 release

File: src/mngt/ocf_mngt_cache.c [hunk 1]
```diff
@@ -2013,11 +2035,11 @@ void ocf_mngt_cache_stop(ocf_cache_t cache,
 
 	OCF_CHECK_NULL(cache);
 
-	/*
-	 * FIXME: What if creating/setting management queue failed?
```
[the remark this answers is not in the document]
Comment (arutk) [reply src/mngt/ocf_mngt_cache.c]: Why do you think it is not fixed? I changed mngt_queue check below to ENV_BUG_ON to make it clear that mngt_queue must be set when cache is attached.

[the remark this answers is not in the document]
Comment (robertbaldyga) [reply src/mngt/ocf_mngt_cache.c]: This is not correct solution. BUG_ON should be used only to mark unrecoverable conditions that normally would never happen. User should not be able to trigger BUG_ON using public API functions. If API is used incorrectly (parameters are out of range, call is in wrong context, etc.) we should return proper error code, never break execution with BUG_ON.

[the remark this answers is not in the document]
Comment (arutk) [reply src/mngt/ocf_mngt_cache.c]: How exactly do you think user would trigger this BUG_ON?

**Pull request merged**